stock_warehouse_orderpoint_stock_info/models/stock_warehouse_orderpoint.py

```diff
@@ -14,14 +14,18 @@ class StockWarehouseOrderpoint(models.Model):
 
     @api.multi
     def _compute_product_available_qty(self):
-        for rec in self:
-            product_available = rec.product_id.with_context(
-                location=rec.location_id.id
-                )._product_available()[rec.product_id.id]
-            rec.product_location_qty = product_available['qty_available']
-            rec.incoming_location_qty = product_available['incoming_qty']
-            rec.outgoing_location_qty = product_available['outgoing_qty']
-            rec.virtual_location_qty = product_available['virtual_available']
+        for loc in self.mapped('location_id'):
```

Comment:
@fmdl IMO we should avoid the call to `mapped`followed by a `filtered`. All you need is to process the list of orderpoint by location.
```python
from  collection import defaultdict
op_by_loc = defaultdict(set)
for rec in self:
     op_by_loc(rec.location_id).add(rec)
for location_id, order_in_loc in op_by_loc.items():
    products = order_in_loc.mapped('product_id').with_context(
        location=loc.id)._compute_quantities_dict(lot_id=None, owner_id=None, package_id=None)
    for rec in order_in_loc:
        ...
```